code-studio/src/js/progress.js

```diff
@@ -157,13 +158,23 @@ function loadProgress(scriptData, currentLevelId, saveAnswersBeforeNavigation =
         newProgress[key] = clientState.mergeActivityResult(state.progress[key], action.progress[key]);
       });
 
+      const stages = state.stages.map(stage => Object.assign({}, stage, {levels: stage.levels.map((level, index) => {
+        const id = level.uid || progress.bestResultLevelId(level.ids, newProgress);
+
+        if (action.peerReviewsPerformed && stage.flex_category === 'Peer Review') {
+          Object.assign(level, action.peerReviewsPerformed[index]);
+        }
+
+        return Object.assign({}, level, {
+          status: level.kind === 'peer_review' ? level.status : progress.activityCssClass(newProgress[id]),
+          id: id,
+          url: level.url
+        });
+      })}));
```

Comment:
FYI, I have an open PR that might result in some merge conflicts with this. https://github.com/code-dot-org/code-dot-org/pull/9379